src/composables/auth.ts

```diff
@@ -256,7 +272,11 @@ export const useAuth = createCustomScopedComposable(() => {
         if (isBiometricLoginEnabled.value && await checkBiometricLoginAvailability()) {
           await openModal(MODAL_BIOMETRIC_LOGIN);
         }
-      } else if (!encryptionKey.value) {
+      } else if (
+        !encryptionKey.value
+        && mnemonic.value
+        // Do not show `openPasswordLoginModal` if user currently initiating password
+        && !modalsOpen.value.find(({ name }) => name === MODAL_SET_PASSWORD)) {
```

Comment:
I think we should not decide if password needs to be requested based on the open modals.